numba/typed/typedobjectutils.py

```diff
@@ -197,3 +197,80 @@ def build_wrapper(fn):
     builder.ret(context.get_constant(types.int32, -1))
 
     return equal_fn
+
+
+def _get_primitive_equal(context, module, datamodel, func_name_suffix):
```

Comment:
If I understood correctly, this function would be equivalent to load two pointers and compare if their contents are equal? What is the input to those functions? Is the dictionary itself?

Reply:
here: 
```c
/* Returns -1 for error; 0 for not equal; 1 for equal */
static int
key_equal(NB_DictKeys *dk, const char *lhs, const char *rhs) {
    if ( dk->methods.key_equal ) {
        return dk->methods.key_equal(lhs, rhs);
    } else {
        return memcmp(lhs, rhs, dk->key_size) == 0;
    }
}
```
input arguments are `lhs` and `rhs` if we define a `key_equal` function in LLVM IR module.